recipes/node-url-to-whatwg-url/src/url-parse.ts

```diff
@@ -1,37 +1,49 @@
-import { getNodeImportStatements } from "@nodejs/codemod-utils/ast-grep/import-statement";
-import { getNodeRequireCalls } from "@nodejs/codemod-utils/ast-grep/require-call";
-import { resolveBindingPath } from "@nodejs/codemod-utils/ast-grep/resolve-binding-path";
-import type { SgRoot, Edit } from "@codemod.com/jssg-types/main";
-import type JS from "@codemod.com/jssg-types/langs/javascript";
+import { getNodeImportStatements } from '@nodejs/codemod-utils/ast-grep/import-statement';
+import { getNodeRequireCalls } from '@nodejs/codemod-utils/ast-grep/require-call';
+import { resolveBindingPath } from '@nodejs/codemod-utils/ast-grep/resolve-binding-path';
+import type { SgRoot, Edit } from '@codemod.com/jssg-types/main';
+import type JS from '@codemod.com/jssg-types/langs/javascript';
 
 const validVariableNameRegex = /^[$A-Z_a-z][$\w]*$/;
 const destructuringAssignmentRegex = /^[^{]+{\s*[^}]+\s*}\s*=\s*/;
 const fieldsToReplace = [
-		{
-			key: "auth",
-			replaceFn: (base: string, hadSemi: boolean, declKind: "const" | "let" | "var") => {
-				const kind = declKind === "var" ? "let" : declKind;
-
-				return `${kind} auth = \`\${${base}.username}:\${${base}.password}\`${hadSemi ? ";" : ""}`;
-			},
+	{
+		key: 'auth',
+		replaceFn: (
+			base: string,
+			hadSemi: boolean,
+			declKind: 'const' | 'let' | 'var',
+		) => {
+			const kind = declKind === 'var' ? 'let' : declKind;
+
+			return `${kind} auth = \`\${${base}.username}:\${${base}.password}\`${hadSemi ? ';' : ''}`;
 		},
-		{
-			key: "path",
-			replaceFn: (base: string, hadSemi: boolean, declKind: "const" | "let" | "var") => {
-				const kind = declKind === "var" ? "let" : declKind;
-
-				return `${kind} path = \`\${${base}.pathname}\${${base}.search}\`${hadSemi ? ";" : ""}`;
-			},
+	},
+	{
+		key: 'path',
+		replaceFn: (
+			base: string,
+			hadSemi: boolean,
+			declKind: 'const' | 'let' | 'var',
+		) => {
+			const kind = declKind === 'var' ? 'let' : declKind;
+
+			return `${kind} path = \`\${${base}.pathname}\${${base}.search}\`${hadSemi ? ';' : ''}`;
 		},
-		{
-			key: "hostname",
-			replaceFn: (base: string, hadSemi: boolean, declKind: "const" | "let" | "var") => {
-				const kind = declKind === "var" ? "let" : declKind;
-
-				return `${kind} hostname = ${base}.hostname.replace(/^\\[|\\]$/, '')${hadSemi ? ";" : ""}`;
-			},
+	},
+	{
+		key: 'hostname',
+		replaceFn: (
+			base: string,
+			hadSemi: boolean,
+			declKind: 'const' | 'let' | 'var',
+		) => {
+			const kind = declKind === 'var' ? 'let' : declKind;
+
+			return `${kind} hostname = ${base}.hostname.replace(/^\\[|\\]$/, '')${hadSemi ? ';' : ''}`;
```

Comment:
nit: this regex can be extracted/moved above with the others to save re-constructing it every invocation.